CodeEdit/Features/CommandsOverlay/Views/CommandsOverlayView.swift

```diff
@@ -1,78 +1,75 @@
 //
-//  CommandPaletteView.swift
+//  CommandsOverlayView.swift
 //  CodeEdit
 //
 //  Created by Alex Sinelnikov on 24.05.2022.
 //
 
 import SwiftUI
 
-/// Command palette view
-struct CommandPaletteView: View {
+/// Commands overlay view
+struct CommandsOverlayView: View {
 
     @Environment(\.colorScheme)
     private var colorScheme: ColorScheme
 
     @ObservedObject
-    private var state: CommandPaletteViewModel
-
-    @ObservedObject
-    private var commandManager: CommandManager = .shared
+    private var state: CommandsOverlayViewModel
 
     @State
     private var monitor: Any?
 
-    @State
-    private var selectedItem: Command?
+    private let closeOverlay: () -> Void
 
-    private let closePalette: () -> Void
-
-    init(state: CommandPaletteViewModel, closePalette: @escaping () -> Void) {
+    init(state: CommandsOverlayViewModel, closeOverlay: @escaping () -> Void) {
         self.state = state
-        self.closePalette = closePalette
-        state.filteredCommands = commandManager.commands
-    }
-
-    func callHandler(command: Command) {
-        closePalette()
-        command.closureWrapper.call()
-        selectedItem = nil
-        state.commandQuery = ""
-        state.filteredCommands = []
-    }
-
-    func onQueryChange(text: String) {
-        state.commandQuery = text
-        state.fetchMatchingCommands(val: text)
+        self.closeOverlay = closeOverlay
     }
 
     var body: some View {
-        OverlayView<SearchResultLabel, EmptyView, Command>(
+        OverlayView<CommandsOverlayItem, EmptyView, CodeEditCommand>(
             title: "Commands",
             image: Image(systemName: "magnifyingglass"),
-            options: $state.filteredCommands,
+            options: $state.filteredMenuCommands,
             text: $state.commandQuery,
             alwaysShowOptions: true,
             optionRowHeight: 30
-        ) { command in
-            SearchResultLabel(labelName: command.title, textToMatch: state.commandQuery)
-        } onRowClick: { command in
-            callHandler(command: command)
+        ) { command, selected in
+            CommandsOverlayItem(command: command, textToMatch: state.commandQuery, selected: selected)
+        } onRowClick: {
+            $0.runAction()
         } onClose: {
-            closePalette()
+            closeOverlay()
         }
         .onReceive(state.$commandQuery.debounce(for: 0.2, scheduler: DispatchQueue.main)) { _ in
```

Comment:
This debounce makes this feel laggy. The search is actually running *really* fast (like a couple ms when I tested) and removing this debounce makes the view feel much more snappy.

Reply:
The debounce needs to be there, otherwise the UI is always one step behind your input text.